web3deployment/retrieveArtifacts.js

```diff
@@ -0,0 +1,37 @@
+const fs = require('fs');
+const util = require('util');
+const readdir = util.promisify(fs.readdir);
+const path = require("path");
+const artifactsPath = path.join(__dirname, "../artifacts/");
+const execSync = require('child_process').execSync;
+
+module.exports.retrieveArtifacts = main;
+async function main() {
+  if (fs.existsSync(artifactsPath)) {
+    let output = await packageArtifacts();
+    return output;
+  } else {
+    compileArtifacts();
```

Comment:
Nice.
So this is much simpler than we had.
Now I understand your comment about it.

I think correct function name is compile contracts.

artifacts is result. 
no?